src/cache.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * rv32emu is freely redistributable under the MIT License. See the file
+ * "LICENSE" for information on usage and redistribution of this file.
+ */
+
+#pragma once
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "list.h"
+#define CACHE_INFO 0
+
+typedef enum { LRU_list, LFU_list, LRU_ghost_list, LFU_ghost_list } list_type_t;
+
+struct hashtable;
+
+typedef struct cache {
```

Comment:
Make use of forward declarations, so that we don't have to disclose the implementation details.